mobile/Ihc/services/ServerDataService.js

```diff
@@ -279,9 +297,11 @@ export function getMedications(name) {
       return Promise.reject(err);
     });
 }
+
 // Server endpoint: put /medication-inventory/:name/update/:date
-export function updateMedication(name, update) {
-  return fetch(fetchUrl + '/medication-inventory/' + name + '/update', {
+// NOTE: separate update and create medication functions in order to check for pre-existing(create) and non-existing(update)
+export function updateMedication(oldKey, update) {
```

Comment:
Just to reiterate the problems with changing the keys, these server functions will break if you change the key